src/vs/workbench/parts/preferences/browser/keybindingWidgets.ts

```diff
@@ -219,26 +227,46 @@ export class DefineKeybindingWidget extends Widget {
 		}));
 
 		this._keybindingInputWidget = this._register(this.instantiationService.createInstance(KeybindingInputWidget, this._domNode.domNode, {}));
-		this._register(this._keybindingInputWidget.onKeybinding(keybinding => this.printKeybinding(keybinding)));
+		this._register(this._keybindingInputWidget.onKeybinding(keybinding => { this.printKeybinding(keybinding); this._onDidChange.fire(keybinding); }));
 		this._register(this._keybindingInputWidget.onEnter(() => this.hide()));
 		this._register(this._keybindingInputWidget.onEscape(() => this.onCancel()));
 		this._register(this._keybindingInputWidget.onBlur(() => this.onCancel()));
 
 		this._outputNode = dom.append(this._domNode.domNode, dom.$('.output'));
+		this._conflictNode = dom.append(this._domNode.domNode, dom.$('.conflicts'));
 	}
 
 	private printKeybinding(keybinding: [ResolvedKeybinding, ResolvedKeybinding]): void {
 		const [firstPart, chordPart] = keybinding;
 		this._firstPart = firstPart;
 		this._chordPart = chordPart;
 		dom.clearNode(this._outputNode);
+		dom.clearNode(this._conflictNode);
 		new KeybindingLabel(this._outputNode, OS).set(this._firstPart, null);
 		if (this._chordPart) {
 			this._outputNode.appendChild(document.createTextNode(nls.localize('defineKeybinding.chordsTo', "chord to")));
 			new KeybindingLabel(this._outputNode, OS).set(this._chordPart, null);
 		}
 	}
 
+	public printConflicts(numConflicts: number, keybinding: [ResolvedKeybinding, ResolvedKeybinding]): void {
```

Comment:
`keybinding` argument is not used, so this can be removed.

Reply:
@sandy081 Within this method, I am using the `keybinding` argument for the event listener for when the link I created is clicked. Please see line 267.

Since I generate the link within the method, I believe I need the argument here to pass the proper keybinding to the event such that it can be passed on to `showSimilarKeybindings` in `keybindingsEditor.ts`, in order to display the matching keybinding entries.

Reply:
@brandonrninefive My suggestion is as follows:

- Change `onDidChange` and `onLinkClick` events to emit key string just like how define method returns key string 

- While you are triggering `onLinkClick` you can get the key from the current state `this._firstPart` and `this._chordPart`. See `define()` method for how to get the key